cmd/fund/fund.go

```diff
@@ -0,0 +1,132 @@
+	nonce, err := web3Client.Eth.GetNonce(senderAddress, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, wallet := range wallets {
+		_, err := web3Client.Eth.SyncSendRawTransaction(
+			common.HexToAddress((*wallet).Hex()),
+			amountWei,
+			nonce,
+			walletFundingGas,
+			web3Client.Utils.ToGWei(1),
+			nil,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("Funded", (*wallet).Hex(), "with", amountWei, "wei")
+		nonce++
+	}
+	return nil
+}
+
+// fundCmd represents the fund command
+var FundCmd = &cobra.Command{
+	Use:   "fund",
+	Short: "Bulk fund many crypto wallets automatically.",
+	Long:  usage,
+	Run: func(cmd *cobra.Command, args []string) {
+		fundingErr := runFunding(cmd)
+		if fundingErr != nil {
+			cmd.PrintErrf("Error funding wallets")
+		}
+	},
+}
+
+func runFunding(cmd *cobra.Command) error {
+	// setup new web3 session with remote rpc node
+	web3Client, clientErr := web3.NewWeb3(chainRPC)
+	if clientErr != nil {
+		cmd.PrintErrf("There was an error creating web3 client: %s", clientErr.Error())
+		return clientErr
+	}
+
+	// add pk to session for sending signed transactions
+	if setAcctErr := web3Client.Eth.SetAccount(fundingWalletPK); setAcctErr != nil {
+		cmd.PrintErrf("There was an error setting account with pk: %s", setAcctErr.Error())
+		return setAcctErr
+	}
+
+	// set proper chainId for corresponding chainRPC
+	cdkChainId := int64(chainID)
+	web3Client.Eth.SetChainId(cdkChainId)
+
+	// convert wallet address and pk format for downstream processing
+	fundingWalletAddressParsed := common.HexToAddress(fundingWalletAddress)
+	fundingWalletECDSA, ecdsaErr := crypto.HexToECDSA(fundingWalletPK)
+	if ecdsaErr != nil {
+		cmd.PrintErrf("There was an error getting ECDSA: %s", ecdsaErr.Error())
+		return ecdsaErr
+	}
+
+	// generate set of new wallet addresses
+	addresses, genWalletErr := generateWalletAddresses(walletCount)
+	if genWalletErr != nil {
+		cmd.PrintErrf("There was an error generating wallet addresses: %s", genWalletErr.Error())
+		return genWalletErr
+	}
+
+	// fund all crypto wallets
+	fmt.Println("Starting to fund loadtest wallets...")
+	fundWalletErr := fundWallets(web3Client, addresses, fundingWalletAddressParsed, fundingWalletECDSA, big.NewInt(int64(walletFundingAmt*1e18)), uint64(walletFundingGas))
+	if fundWalletErr != nil {
+		cmd.PrintErrf("There was an error funding wallets: %s", fundWalletErr.Error())
+		return fundWalletErr
+	}
+
+	return nil
+}
+
+func init() {
+	FundCmd.Flags().IntVar(&walletCount, "wallet-count", 2, "Number of wallets to fund")
+	FundCmd.Flags().StringVar(&fundingWalletAddress, "funding-wallet-address", "", "Origin wallet that will be doing the funding")
+	FundCmd.Flags().StringVar(&fundingWalletPK, "funding-wallet-pk", "", "Corresponding private key for funding wallet address, ensure you remove leading 0x")
+	FundCmd.Flags().IntVar(&chainID, "chain-id", 100, "Blockchain network chain id")
```

Comment:
My preference here would be to just drop this flag unless there is a reason why a user would want to specify a chain id. `loadtest` used to have this flag, but as long as your RPC supports returning the chain id, i think that's easier to manage

Reply:
i find this is required. if user does not specify correct chain id, sendRawTransaction throws 'invalid sender' errors i believe, ran into this myself

Reply:
you can probably reuse this logic in `rpcfuzz` to prevent the usage of flags: https://github.com/maticnetwork/polygon-cli/blob/feat/revert-msg-conformance/cmd/rpcfuzz/rpcfuzz.go#L220-L224

Reply:
@IdrisHanafi seeing: 
<img width="635" alt="image" src="https://github.com/maticnetwork/polygon-cli/assets/100106211/f8fb2f33-a469-4f9e-940a-7421201bda29">

Reply:
You will definitely need a chain id, but you can determine the chain id with an rpc call. If the rpc itself tells you the chain id, it's going to be very unusual that you need to specify the chain id manually. The reasons that I can think of would be to avoid making an extra RPC call (not a big deal in the case) or for some reason the chain id doesn't match or the rpc is failing (ideally that's fixed in the rpc itself).

https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_chainid
Here is an example from `loadtest`
https://github.com/maticnetwork/polygon-cli/blob/601409d7edc2bdfc5736eff6b974828e18e77767/cmd/loadtest/loadtest.go#L201-L206